parts/k8s/manifests/kubernetesmaster-kube-controller-manager.yaml

```diff
@@ -11,25 +11,46 @@ spec:
   hostNetwork: true
   containers:
     - name: kube-controller-manager
-      image: {{GetHyperkubeImageReference}}
+      image: {{ContainerImage "kube-controller-manager"}}
       imagePullPolicy: IfNotPresent
-      command: ["/hyperkube", "kube-controller-manager"]
-      args: [{{GetK8sRuntimeConfigKeyVals .Properties.OrchestratorProfile.KubernetesConfig.ControllerManagerConfig}}]
+      command: [{{ContainerConfig "command"}}]
+      args: [{{GetControllerManagerArgs}}]
+{{- if IsAzureStackCloud}}
+      env:
+      - name: AZURE_ENVIRONMENT_FILEPATH
+        value: "/etc/kubernetes/azurestackcloud.json"
+{{end}}
       volumeMounts:
         - name: etc-kubernetes
           mountPath: /etc/kubernetes
+{{- if IsKubernetesVersionGe "1.17.0"}}
```

Comment:
One advantage of keeping all the separate manifests was that it made it easier to diff them with the canonical upstream versions. Generally we've kept manifests as synced up as possible to facilitate this.

Just sayin', probably not worth changing this PR.

Reply:
Right. This approach just seems so much more maintainable. Perhaps we need UT tooling to do some go template-processed static validation against the upstream sources.

Reply:
Agree that this is more maintainable.